toke.c

```diff
@@ -4493,7 +4493,7 @@ S_intuit_more(pTHX_ char *s, char *e)
 
     /* Keep track of how many multiple occurrences of the same character there
      * are */
-    char seen[256];
+    U8 seen[256];
```

Comment:
This change breaks a test in pat.t, showing that the code worked by happenstance.  That test wanted 2000 'b' characters in a row to be treated as a character class.  No one, outside poor generated code, would write the same character multiple times inside a character class: it would reduce to just the first occurrence.  And the code intended to make such things to be interpreted as subscripts, by penalizing it from being a character class with increasing slope for each added redundant character.  But because this array was declared signed, after you got to 127, the next increment would be to -126 or -127 (which I believe is undefined behavior) and that would **subtract** instead of adding additional penalty.  This would continue to -1, then more undefined behavior to get to 0, and the cycle repeats.  The 2000 was arbitrarily chosen; other repeat factors would have resulted in different results

Reply:
Actually on a twos complement box, it goes from 127 to -128, which will over the course of many cycles will cause the weight to become more and more negative; exactly the opposite of what the code intends.

Reply:
The breakage actually stemmed from a different change, but this remains, and shows the brittleness of the code